tests/python-gpu/test_device_quantile_dmatrix.py

```diff
@@ -32,32 +32,41 @@ def test_dmatrix_cupy_init(self) -> None:
         xgb.DeviceQuantileDMatrix(data, cp.ones(5, dtype=np.float64))
 
     @pytest.mark.skipif(**tm.no_cupy())
-    def test_from_host(self) -> None:
+    @pytest.mark.parametrize(
+        "tree_method,max_bin", [
+            ("hist", 16), ("gpu_hist", 16), ("hist", 64), ("gpu_hist", 64)
+        ]
+    )
+    def test_interoperability(self, tree_method: str, max_bin: int) -> None:
         import cupy as cp
         n_samples = 64
         n_features = 3
         X, y, w = tm.make_batches(
             n_samples, n_features=n_features, n_batches=1, use_cupy=False
         )
-        Xy = xgb.QuantileDMatrix(X[0], y[0], weight=w[0])
-        booster_0 = xgb.train({"tree_method": "gpu_hist"}, Xy, num_boost_round=4)
+        # from CPU
+        Xy = xgb.QuantileDMatrix(X[0], y[0], weight=w[0], max_bin=max_bin)
```

Comment:
This is very nice. The interface just works and there is no confusion for users.